### ISSUE TITLE
Raise an error on selecting the path shown in the bottom

### ISSUE THREAD

**Issue by 6br** [vgteam/sequenceTubeMap#20]
Hello, I am grateful to this library and I would like to use it to display genome graphs :)

I found that it causes an error when I click the path shown in the bottom. In this case, I received an error when I make `ref` unchecked.
![2017-10-30 18 07 28](https://user-images.githubusercontent.com/12047794/32162714-4297b1f8-bd9d-11e7-8c3f-fbdee1812895.png)

And I have several questions:

* Could it visualize not only paths but also nodes and edges without path information?
* Are there onClickNode callback or onClickPath callback as external API?

Thank you,

**Comment by wolfib** [comment]
Hi, thanks for your interest in this project!

Thanks for telling me about that bug. I could reproduce it and I will look for what is causing it.

For now, visualizing just a graph without path info is not possible. You are not the only one who would like this feature though and it is something that might come in the (non-immediate) future. Not sure how to best approach this and will have to give it some more thought first.

External API callbacks sound interesting. Do I understand this correctly? You would want the ability to somehow register your own functions to be called on click on a node/path? Since external code does not know the names of nodes/tracks this would probably need to be triggered with a click on any element and not a specific element. Do you have any other projects in mind that has such a functionality?

**Comment by 6br** [comment]
Thanks for your reply!

Yes, I would like the ability to register my own functions to be called on click on a node/path. And it should return the name of nodes/tracks to be able to extract the information such as a sequence of node or path since external codes do not know the name.

And I have a concept of the "graph genome browser" in mind. I would appreciate it if you have ideas or suggestions for integrations with other components such as the input box of genomic coordinates or gene name and information about extracted paths or nodes.

Regards,

**Comment by wolfib** [comment]
Exactly, a graph genome browser is the vision I have in mind too. For the moment, I am mostly focusing on making it easier to build/use and to use your own data before adding new features.

**Comment by adamnovak** [comment]
This might be fixed, since a bunch of the code has changed since it was reported. We should see if we can reproduce it with the example data.
